### ISSUE TITLE
Error: Variable "$foo" is never used in operation "FooQuery"

### ISSUE THREAD

**Issue by edorivai** [dotansimha/graphql-code-generator#726]
**Describe the bug**
As described in https://github.com/dotansimha/graphql-code-generator/pull/628#issuecomment-423914784, when using a variable **only** inside a fragment that lives in another file, the generator will yield _unused variable_ validation errors:

```
Error: Variable "$foo" is never used in operation "FooQuery"
```

**To Reproduce**
(taken from: https://github.com/dotansimha/graphql-code-generator/pull/628#issuecomment-424015962):

Here is a somewhat contrived example:

**file1.ts**

```ts
import {barFragment} from "./file2";

export const query = gql`
  query FooQuery($selectedBars: [ID!]!) {
     ...Bar_query
  }
  ${barFragment}
 `;
```

**file2.ts**

```ts
export const barFragment = gql`
    fragment Bar_query on Query {
      bars($selectedBars) {
        id
     }
  `;
```

**Expected behavior**
No validation errors

**Schema/Documents**
[This is the part of the GraphQL spec](https://facebook.github.io/graphql/June2018/#sec-All-Variables-Used) that explicitly outlines that `UnusedVariable` validation should take fragments into account.

**Comment by dotansimha** [comment]
I think we can ignore this validation as well. I'll take a look.

**Comment by dotansimha** [comment]
@edorivai  available in 0.13.0.
